agent/grpc/client.go

```diff
@@ -61,8 +61,7 @@ func (c *ClientConnPool) ClientConn(datacenter string) (*grpc.ClientConn, error)
 		grpc.WithInsecure(),
 		grpc.WithContextDialer(c.dialer),
 		grpc.WithDisableRetry(),
-		// TODO: previously this statsHandler was shared with the Handler. Is that necessary?
```

Comment:
I'm pretty sure that using separate statsHandlers is correct. We don't want to share a single counter for both client and server connections because those are emitted as separate metrics.